janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/JointIndexQuery.java

```diff
@@ -74,8 +74,9 @@ public boolean isEmpty() {
     }
 
     @Override
-    public void observeWith(QueryProfiler profiler) {
-        queries.forEach(q -> q.observeWith(profiler));
+    public void observeWith(QueryProfiler profiler, boolean hasSiblings) {
+        boolean isSingleQuery = queries.size() == 1;
```

Comment:
Not sure that `consistsOfMultipleQueries` is a right name because it will be `true` when it consists of 0 queries as well.
My point was that the variable is always inverted when it is used. So, I thought it will be easier if we don't invert it below. 
That said, I'm fine leaving it as is

Reply:
My initial thought was "isSingleQuery" might be easier to understand, but since doing so require multiple additional inversion operations, let me change it.
I can change it to:
`boolean consistsOfMultipleQueries = queries.size() > 1;`

Reply:
Got it, thanks. Just want to notice that `boolean isNotSingleQuery = queries.size() != 1` and `boolean consistsOfMultipleQueries = queries.size() > 1;` are different because the first one will return `true` for `0` but second will return `false` for `0`.